alternator/executor.cc

```diff
@@ -234,6 +235,18 @@ static std::string get_table_name(const rjson::value& request) {
     return *name;
 }
 
+executor::executor(gms::gossiper& gossiper,
+    service::storage_proxy& proxy,
+    service::migration_manager& mm,
+    db::system_distributed_keyspace& sdks,
+    cdc::metadata& cdc_metadata,
+    smp_service_group ssg,
+    utils::updateable_value<uint32_t> default_timeout_in_ms)
+        : _gossiper(gossiper), _proxy(proxy), _mm(mm), _sdks(sdks), _cdc_metadata(cdc_metadata),
+          _ssg(ssg), _expression_cache(make_lw_shared<expression_cache_t>(2048, 60min, elogger)) {
```

Comment:
It makes sense to throw away parsed expressions that aren't reused in a timely manner. Otherwise they just sit there forever consuming space.

Reply:
Maybe, but I don't think it's worth the hassle (and I don't know how much runtime effort, and how much code, is involved in loading_cache to keep that timeout thing).
For example, if we just keep the last few expressions for each table, I think this would cover most use cases and the memory will be negligible.
I guess that if we do use loading_cache, we might as well use the "60min" option. Maybe.